Support netcoreapp2.1

When I try running a BenchmarkDotNet project that looks like the following:

```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="BenchmarkDotNet" Version="0.10.10" />
  </ItemGroup>
</Project>
```

I get the errors:

```
// Exception:   Restoring packages for /home/eerhardt/DotNetTest/RefOverloadsPerfTests/bin/Release/netcoreapp2.1/7d0199c9-e7ac-414c-b788-16da68c57138/BenchmarkDotNet.Autogenerated.csproj...
/home/eerhardt/DotNetTest/RefOverloadsPerfTests/bin/Release/netcoreapp2.1/7d0199c9-e7ac-414c-b788-16da68c57138/BenchmarkDotNet.Autogenerated.csproj : error NU1201: Project RefOverloadsPerfTests is not compatible with netcoreapp2.0 (.NETCoreApp,Version=v2.0). Project RefOverloadsPerfTests supports: netcoreapp2.1 (.NETCoreApp,Version=v2.1)
  Generating MSBuild file /home/eerhardt/DotNetTest/RefOverloadsPerfTests/bin/Release/netcoreapp2.1/7d0199c9-e7ac-414c-b788-16da68c57138/obj/BenchmarkDotNet.Autogenerated.csproj.nuget.g.props.
  Generating MSBuild file /home/eerhardt/DotNetTest/RefOverloadsPerfTests/bin/Release/netcoreapp2.1/7d0199c9-e7ac-414c-b788-16da68c57138/obj/BenchmarkDotNet.Autogenerated.csproj.nuget.g.targets.
  Restore failed in 131.93 ms for /home/eerhardt/DotNetTest/RefOverloadsPerfTests/bin/Release/netcoreapp2.1/7d0199c9-e7ac-414c-b788-16da68c57138/BenchmarkDotNet.Autogenerated.csproj.
```

I'm using the latest .NET Core SDK from the master branch:  https://github.com/dotnet/cli#installers-and-binaries and I have a NuGet.config that points to the daily builds of .NET Core:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
      <add key="nuget" value="https://api.nuget.org/v3/index.json" />
      <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
  </packageSources>
</configuration>
```

I expected this to just work, like it does for `netcoreapp2.0`.

--- a/src/BenchmarkDotNet.Core/Toolchains/DotNetCli/NetCoreAppSettings.cs
+++ b/src/BenchmarkDotNet.Core/Toolchains/DotNetCli/NetCoreAppSettings.cs
@@ -12,6 +12,7 @@ namespace BenchmarkDotNet.Toolchains.DotNetCli
         [PublicAPI] public static readonly NetCoreAppSettings NetCoreApp11 = new NetCoreAppSettings("netcoreapp1.1", null, ".NET Core 1.1");
         [PublicAPI] public static readonly NetCoreAppSettings NetCoreApp12 = new NetCoreAppSettings("netcoreapp1.2", null, ".NET Core 1.0");
         [PublicAPI] public static readonly NetCoreAppSettings NetCoreApp20 = new NetCoreAppSettings("netcoreapp2.0", null, ".NET Core 2.0");
+        [PublicAPI] public static readonly NetCoreAppSettings NetCoreApp21 = new NetCoreAppSettings("netcoreapp2.1", null, ".NET Core 2.1");
 
         private static NetCoreAppSettings Default =>
 #if NETCOREAPP2_0
@@ -82,8 +83,10 @@ namespace BenchmarkDotNet.Toolchains.DotNetCli
                 // the problem is that both netcoreapp1.2 and netcoreapp2.0 have 
                 // "System.Runtime Version=1.2.0.0". 
                 // 2.0 was officialy announced name, so let's bet on it (1.2 was probably an internal dev thing)
-                if (assembly.FullName.Contains("Version=4.2"))
+                if (assembly.FullName.Contains("Version=4.2.0"))
                     return NetCoreApp20;
+                if (assembly.FullName.Contains("Version=4.2.1"))
+                    return NetCoreApp21;
             }
             catch
             {

